### PR TITLE
add approve function

### PR DESCRIPTION
**Pull request by sriyantra**

### CODE REVIEW

File: contracts/CErc20PluginRewardsDelegate.sol [hunk 1]
```diff
@@ -11,18 +11,22 @@ contract CErc20PluginRewardsDelegate is CErc20PluginDelegate {
...
     }
 
     /// @notice A reward token claim function 
     /// to be overriden for use cases where rewardToken needs to be pulled in
     function claim() external {}
-}
\ No newline at end of file
+
+    /// @notice token approval function
+    function approve(address _token, address _spender) external {
```
Comment (Joeysantoro) [inline contracts/CErc20PluginRewardsDelegate.sol:28]: Add require token != underlying and token != plugin

File: contracts/CErc20RewardsDelegate.sol [hunk 2]
```diff
@@ -5,22 +5,14 @@ import "./CErc20Delegate.sol";
...
+    /// @notice A reward token claim function 
+    /// to be overriden for use cases where rewardToken needs to be pulled in
+    function claim() external {} 
 
-        (address _rewardsDistributor, address _rewardToken) = abi.decode(
-            data,
-            (address, address)
-        );
+    /// @notice token approval function
+    function approve(address _token, address _spender) external {
```
Comment (Joeysantoro) [inline contracts/CErc20RewardsDelegate.sol:13]: require token != asset

File: contracts/CErc20PluginRewardsDelegate.sol (hunk 1 above)
Comment (sriyantra) [reply contracts/CErc20PluginRewardsDelegate.sol:27]: we have to approve plugin to spend underlying though right

Comment (Joeysantoro) [reply contracts/CErc20PluginRewardsDelegate.sol:28]: should `or` this with token != plugin so the plugin shares can't be approved/withdrawn

**Review by Joeysantoro — CHANGES_REQUESTED** [review]
one last change and good to go

File: contracts/CErc20PluginRewardsDelegate.sol (hunk 1 above)
Comment (sriyantra) [reply contracts/CErc20PluginRewardsDelegate.sol:28]: updated

**Pull request merged**